addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -62,4 +69,37 @@ def _process_step_create_lead(self, discuss_channel):
         create_values.update(self._chatbot_crm_prepare_lead_values(
             discuss_channel, customer_values['description']))
 
-        self.env['crm.lead'].create(create_values)
+        return self.env['crm.lead'].create(create_values)
+
+    def _process_step_create_lead_and_forward(self, discuss_channel):
+        lead = self._process_step_create_lead(discuss_channel)
+        team = self.crm_team_id
+        domain = literal_eval(team.assignment_domain or '[]')
+        lead = lead.filtered_domain(domain)
+        if team and lead:
+            member_to_assign = next(
+                (
+                    member for member in team.crm_team_member_ids
+                    if not member.assignment_optout
+                    and member._get_assignment_quota() > 0
+                ),
+                None
+            )
+            if member_to_assign:
+                lead.user_id = member_to_assign.user_id
+                if member_to_assign.user_id._is_user_available():
+                    user = member_to_assign.user_id
+                    discuss_channel.with_user(user).sudo().add_members(user.partner_id.ids, open_chat_window=True)
```

Comment:
Provide a reason for using `sudo` here as a comment.

Reply:
You can store the sudo channel and use it directly rather than doing sudo again.